lib/byron/src/Cardano/Wallet/Byron.hs

```diff
@@ -200,22 +202,23 @@ serveWallet
     serveApp socket = do
         withNetworkLayer nullTracer bp addrInfo versionData $ \nl -> do
             withNtpClient ntpClientTracer ntpSettings $ \ntpClient -> do
-                byronApi   <- apiLayer (newTransactionLayer @n) nl
-                icarusApi  <- apiLayer (newTransactionLayer @n) nl
-                startServer socket byronApi icarusApi ntpClient $> ExitSuccess
+                let pm = fromNetworkMagic $ networkMagic $ fst versionData
+                randomApi  <- apiLayer (newTransactionLayer @n pm) nl
+                icarusApi  <- apiLayer (newTransactionLayer @n pm) nl
+                startServer socket randomApi icarusApi ntpClient $> ExitSuccess
 
     startServer
         :: Socket
         -> ApiLayer (RndState 'Mainnet) t ByronKey
         -> ApiLayer (SeqState 'Mainnet IcarusKey) t IcarusKey
         -> NtpClient
         -> IO ()
-    startServer socket byron icarus ntp = do
+    startServer socket random icarus ntp = do
```

Comment:
I understand (or I hope to understand) the change here - we have Byron era, and in Byron era a number of things can be supported random, icarus... But is not icarus using random scheme?

Reply:
> we have Byron era, and in Byron era a number of things can be supported random, icarus...

Correct :+1: 

> .. But is not icarus using random scheme? 

Incorrect :sweat_smile: 

> Any chance to put somewhere documentation about differences between random and icarus and how they fit within Byron - or maybe link to the source describing this?

Here: https://github.com/input-output-hk/cardano-wallet/wiki/About-Address-Derivation#differences-between-hd-random-and-hd-sequential-wallets

and here: https://github.com/input-output-hk/cardano-wallet/wiki/Wallet-Cryptography-and-Encoding#master-key-generation

Reply:
(not that I renamed this to indeed avoid confusion between the Byron era which so to speak includes Icarus wallets and Random wallets which are anyway part of the Shelley era  too :upside_down_face: )